### ISSUE TITLE
Import plant:method and generator:method

### ISSUE THREAD

**Issue by Gazer75** [openinframap/openinframap#46]
Adding this to tags could make it possible to differentiate hydro power plants by "run of the river" and storage.
Right now one would have to somehow match the generators to the plant to find this.

**Comment by flacombe** [comment]
Which value should be used for plant:method if a given plant involve hydroelectric turbines and photovoltaic panels ?
Method sounds to be a device (generator) property, not a site (plant) attribute

**Comment by Gazer75** [comment]
You describe the same problem as plant:source which is solved by semicolon separation ;)

**Comment by flacombe** [comment]
Which is not desirable
I'm not happy with redundancy implied by device properties mentioned on surrounding site features

**Comment by Gazer75** [comment]
How do you propose to identify the different methods?

Site relations for reservoir based plants is not a good idea. A relation do not define a "center" point so the center would be in the middle of all members. Some hydro plants can create a 100 square km area if you include all dams and tunnels.

**Comment by flacombe** [comment]
Stating methods at the generator level is the less redundant way of mapping. The need to get this data at the plant level is a consumer need and mappers shouldn't deal with this (as they aren't not supposed to understand the need to make device properties redundant at the site level)
Using plant:method would be more acceptable from a QA point of view to check if generator:method and plant:method are consistent... why not, I still need to be convinced with this

Reservoir based plants don't necessarily need a centre
This would be good to render reservoir, dams, tunnels and power houses only

**Comment by Gazer75** [comment]
We can't just ignore the consumers. This is a two way thing. If we make it hard for consumers to use the data they will simply ignore it and go elsewhere.

So you think it is good for a map like this to display a power plant center several kilometers away from the power house/generator hall?

**Comment by russss** [comment]
I struggle to have a strong opinion on this (and it's off-topic here) but I think I agree with @flacombe that it makes more sense on the generator. So I think if I do anything with this tag I'll do `generator:method` first, but I'm not really sure how I'd render it.

**Comment by Gazer75** [comment]
It would be nice to have the method show up when you click on the plant.
How that would be possible without the plant:method I have no idea.

Even searching the OSM db to find water storage vs run-of-the-river is hard without this.

**Comment by flacombe** [comment]
> We can't just ignore the consumers. This is a two way thing. If we make it hard for consumers to use the data they will simply ignore it and go elsewhere.

Be sure I have concerns as a consumer too. My point isn't to ignore consumers, but to map reality
A consumer shouldn't expect mappers to aggregate data according to his own needs

> So you think it is good for a map like this to display a power plant center several kilometers away from the power house/generator hall?

The answer lie in the question: if you assume that power house is the best element to represent a plant, then render it.
Site relations in OSM data don't state what is the center or isn't, or what is important or what isn't. It's only a relation that anyone can process according to his own needs.

In other words, and sorry @russss for the implied overhead : if method of involved generators should be visible on the plant object, OIM will have to pre-process it.

**Comment by russss** [comment]
These have been imported and displayed for a while.
